src/parser.rs

```diff
@@ -239,6 +256,11 @@ impl Parser {
             .collect();
     }
 
+    /// Set the license info
+    pub fn set_license_info(&mut self, license_info: Option<LicenseInfo>) {
+        self.license_info = license_info;
+    }
+
```

Comment:
Slightly better as it allows to pass both an `Option<LicenseInfo>` or directly a `LicenseInfo`:

```rust
pub fn set_license_info<T: Into<Option<LicenseInfo>>>(&mut self, license_info: T) {
    self.license_info = license_info.into();
}
```